tests/ttnn/unit_tests/operations/test_moreh_logsoftmax.py

```diff
@@ -471,29 +518,93 @@ def logsoftmax_backward_step(dev_y, dev_dy, dim, strategy, device, num_program_c
         [[1, 15, 32, 32], 1, ttnn.operations.moreh.SoftmaxBackwardOpParallelizationStrategy.LARGE_C],
     ),
 )
-def test_logsoftmax_backward_callback(shape_dim_strategy, device, use_program_cache):
+@pytest.mark.parametrize(
+    "dtype",
+    [
+        ttnn.bfloat16,
+    ],
+)
+def test_logsoftmax_backward_callback(shape_dim_strategy, dtype, device, use_program_cache):
     shape, dim, strategy = shape_dim_strategy
     torch.manual_seed(0)
 
-    num_program_cache_entries = None
-    for i in range(2):
-        x = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16).requires_grad_(True)
-        y = F.log_softmax(x, dim)
-        dev_y = ttnn.Tensor(y, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
-
-        dy = torch.randint(low=0, high=4, size=shape).to(torch.bfloat16)
-        dev_dy = ttnn.Tensor(dy, ttnn.bfloat16).to(ttnn.TILE_LAYOUT).to(device)
-
-        y.backward(dy)
+    rtol = atol = 0.5
 
-        tt_npu, num_program_cache_entries = logsoftmax_backward_step(
-            dev_y, dev_dy, dim, strategy, device, num_program_cache_entries
+    for i in range(2):
+        run_moreh_logsoftmax_backward_test(
+            shape, dim, dtype, ttnn.TILE_LAYOUT, device, rtol, atol, True, True, strategy=strategy
         )
+        if i == 0:
+            num_program_cache_entries = device.num_program_cache_entries()
+            assert num_program_cache_entries > 0
+        else:
+            assert device.num_program_cache_entries() == num_program_cache_entries
+        torch_dummy = torch.randn([32, 32])
+        tt_dummy = to_ttnn(torch_dummy, device=device)
 
```

Comment:
you can use 
```python
tt_dummy = ttnn.from_torch(torch_dummy, device=device) 
```
and remove the include to_ttnn from all your test files.

Reply:
Resolved